src/services/MatchService.ts

```diff
@@ -0,0 +1,19 @@
+import { PlayerInQueue } from "../repositories/QueueRepository/types";
+import { createRandomTeams } from "../services/TeamAssignmentService";
+import ActiveMatchRepository from "../repositories/ActiveMatchRepository";
+
+export async function createRandomMatch(ballChasers: PlayerInQueue[]): Promise<ReadonlyArray<PlayerInQueue>> {
+  //Assign teams based on MMR
+  const createdTeams = await createRandomTeams(ballChasers);
+
+  //Create Match
+  await ActiveMatchRepository.addActiveMatch(createdTeams);
+
+  ballChasers.forEach((player) => {
```

Comment:
Instead of mutating the parameter, I would just build the list of ballChasers from the createdTeams response. Mutations can get hard to track and cause hard to find bugs so I try and avoid it if possible. That's why the ballChasers type is Readonly. So that it can't be mutated like this.

Reply:
Fixed with a map in next commit.